vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/AthenzSslContextBuilder.java

```diff
@@ -14,20 +15,39 @@
 import java.io.UncheckedIOException;
 import java.security.GeneralSecurityException;
 import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.PrivateKey;
 import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
 
 /**
  * @author bjorncs
  */
 public class AthenzSslContextBuilder {
 
+    public enum KeyStoreType {
+        JKS {
+            KeyStore createKeystore() throws KeyStoreException {
+                return KeyStore.getInstance("JKS");
+            }
+        },
+        PKCS12 {
+            private final BouncyCastleProvider bouncyCastleProvider = new BouncyCastleProvider();
+
+            KeyStore createKeystore() throws KeyStoreException {
```

Comment:
Would it be possible to have this as public? This would allow to remove similar code in node-admin: https://github.com/vespa-engine/vespa/blob/66dfb66beb03b9a979db071fc785295c73ee7962/node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/util/KeyStoreOptions.java#L40

Reply:
If `getKeyStoreInstance ()` is used to construct an `SSLContext`, we should replace it entirely with `AthenzSslContextBuilder`.

Reply:
Its also used to create a new KeyStore so it can be written to disk. https://github.com/vespa-engine/vespa/blob/42124dfaa139428dd1a216422027305bf4ceb5b8/node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/certificate/ConfigServerKeyStoreRefresher.java#L154

Reply:
Okay. I'll do another PR where I split the keystore specific parts out to a separate utility class.